lib/python/ensembl/brc4/runnable/process_gff3.py

```diff
@@ -636,42 +636,110 @@ def build_transcript(self, gene: SeqFeature) -> SeqFeature:
+        sub_cdss = []
+        for subf in mrna.sub_features:
+            if subf.type == "CDS":
+                sub_cdss.append(subf)
+            elif subf.type == "exon":
+                sub_exons.append(subf)
+        
+        if len(sub_cdss) > 0:
+            raise Exception(f"Gene {gene.id} has CDSs as children of the gene and mRNA")
+        if len(sub_exons) > 0:
+            # Check that they match the CDS outside
+            if len(sub_exons) == len(cdss):
+                # Now that all coordinates are the same
+                coord_exons = [ f"{exon.location}" for exon in sub_exons ]
+                coord_cdss = [ f"{cds.location}" for cds in cdss ]
+
+                if coord_exons != coord_cdss:
+                    raise Exception(f"Gene {gene.id} CDSs and exons under the mRNA do not match")
+            else:
+                raise Exception(f"Gene {gene.id} CDSs and exons under the mRNA do not match (different count)")
 
-        # Check that the mRNA does not have CDSs
-        for subm in mrna.sub_features:
-            if subm.type == "CDS":
-                raise Exception(
-                    "Can't move CDS child from gene to mRNA"
-                    f" if mRNA already have some CDS for gene {gene.id}"
-                )
-        mrna.sub_features.append(cds)
+        print(f"Gene {gene.id}: move {len(cdss)} CDSs to the mRNA")
+        # No more issues? move the CDSs
+        mrna.sub_features += cdss
+        # And remove them from the gene
+        gene.sub_features = gene_subf_clean
+        gene.sub_features.append(mrna)
         
-        return mrna
+        return gene
+    
+    def clean_extra_exons(self, gene: SeqFeature) -> SeqFeature:
+        """Remove extra exons, already existing in the mRNA.
+
+        This is a special case where a gene contains proper mRNAs etc. but also
+        extra exons for the same features. Those exons usually have an ID starting with
+        "id-", so that's what we use to detect them.
+        """
+        exons = []
+        mrnas = []
+        others = []
+        for subf in gene.sub_features:
+            if subf.type == "exon":
+                exons.append(subf)
+            elif subf.type == "mRNA":
+                mrnas.append(subf)
+            else:
+                others.append(subf)
+        
+        if exons and mrnas:
+            exon_has_id = 0
+            # Check if the exon ids start with id-, which is an indication that they do not belong here
+            for exon in exons:
+                if exon.id.startswith("id-"):
+                    exon_has_id += 1
+            if exon_has_id:
+                if exon_has_id == len(exons):
+                    print(f"Remove {exon_has_id} extra exons from {gene.id}")
+                    gene.sub_features = mrnas
+                    gene.sub_features += others
+                else:
+                    raise Exception(f"Can't remove extra exons for {gene.id}, some have gene id-, others do not")
```

Comment:
```suggestion
                    raise Exception(f"Can't remove extra exons for {gene.id}, not all start with 'id-'")
```
(to fit it in one single line but adding `'` as needed)